### PR TITLE
Add a remoulade builtin scheduler

### PR DESCRIPTION
**Pull request by mehdigmira**

### CODE REVIEW

File: remoulade/scheduler.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+                self.flush(job_name, job)
+
+    def start(self):
+        self.sync_config()
+
+        while True:
+            now = datetime.datetime.utcnow()
+            curr_isodow = now.isoweekday()
+            with self.client.lock(self.lock_key, timeout=10, blocking_timeout=20):
```
Comment (antoinerabany) [inline remoulade/scheduler.py]: I think I don't get why you use this lock here and not for sync_config ?

File: remoulade/scheduler.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    global global_scheduler
+    if global_scheduler is None:
+        raise ValueError('Scheduler not found, are you sure you called set_scheduler(scheduler) ?')
+    return global_scheduler
+
+
+def set_scheduler(scheduler: "Scheduler"):
+    global global_scheduler
+    global_scheduler = scheduler
+
```
Comment (antoinerabany) [inline remoulade/scheduler.py]: for default vaules, i was putting them a the top like
```
#: The maximum amount of milliseconds results are allowed to exist in
#: the backend.
DEFAULT_RESULT_TTL = 600000
```

File: remoulade/scheduler.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+            last_queued=None if data["last_queued"] is None else datetime.datetime.strptime(data["last_queued"], '%Y-%m-%dT%H:%M:%S.%f'),
+            tz=data["tz"],
+            args=data["args"],
+            kwargs=data["kwargs"]
+        )
+
+
+class Scheduler:
+
+    def __init__(self, broker: Broker, schedule: Dict[str, ScheduledJob], *,
```
Comment (antoinerabany) [inline remoulade/scheduler.py]: Don't forget to add some doc (unit of period for example). And update the changelog.

**Comment by mehdigmira** [comment]
Now TZ are handled

File: tests/test_scheduler.py [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+    with open("/tmp/scheduler-2", "r") as f:
+        text = [x for x in f.read().split("\n") if x]
+
+    # slow task should run exactly once, even if we launched 2 schedulers
+    assert len(text) == 1
+
+
+@pytest.mark.parametrize("suffix", ["", "_tz"])
+def test_scheduler_daily_time(start_scheduler, start_cli, suffix):
+    start_scheduler(f"tests.scheduler_configs.daily{suffix}")
```
Comment (antoinerabany) [inline tests/test_scheduler.py]: we shoule be compatbile 3.5 ?

**Comment by mehdigmira** [comment]
Also add remoulade-ls and remoulade-run commands

**Pull request merged**